src/services/TestNumberService.ts

```diff
@@ -0,0 +1,130 @@
+import {AWSError} from "aws-sdk";
+import {TestNumber} from "../models/TestNumber";
+import {Service} from "../models/injector/ServiceDecorator";
+import {HTTPResponse} from "../utils/HTTPResponse";
+import {DynamoDBService} from "./DynamoDBService";
+
+@Service()
+export class TestNumberService {
+    public readonly dbClient: DynamoDBService;
+
+    /**
+     * Constructor for the TestNumberService class
+     * @param dynamo
+     */
+    constructor(dynamo: DynamoDBService) {
+        this.dbClient = dynamo;
+    }
+
+    /**
+     * Creates a new test number in the database.
+     * @param activity - the payload containing the activity
+     */
+    public async createTestNumber(): Promise<TestNumber> {
+        return this.getLastTestNumber()
+            .then(lastTestNumber => {
```

Comment:
Do we really need to get last 10 numbers ? how are you ensuring this is an atomic operation? What happens when 3 requests hit 3 lambdas , making 3 concurrent calls to database, how do you ensure its still atomic operation?

Reply:
changed it in order to always have only the last value in db. made the write operation not idempotent so we know when two concurrent calls try to generate the same certificate number and implemented retry mechanism on that case